lldb/tools/lldb-dap/src-ts/debug-adapter-factory.ts

```diff
@@ -8,15 +11,7 @@ import { LLDBDapOptions } from "./types";
 export class LLDBDapDescriptorFactory
   implements vscode.DebugAdapterDescriptorFactory
 {
-  private lldbDapOptions: LLDBDapOptions;
-
-  constructor(lldbDapOptions: LLDBDapOptions) {
-    this.lldbDapOptions = lldbDapOptions;
-  }
-
-  static async isValidDebugAdapterPath(
-    pathUri: vscode.Uri,
-  ): Promise<Boolean> {
+  static async isValidFile(pathUri: vscode.Uri): Promise<Boolean> {
```

Comment:
Should this just be a function outside of the class?

Reply:
We're also calling this from `extension.ts`. I guess we could export the function but that doesn't seem much better than leaving it a static method?

Reply:
I don't think this is doing anything specific for `LLDBDapDescriptorFactory` and a top level `isValidExe` or `isExecutable` is straight forward to reason vs `LLDBDapDescriptorFactory.isValidFile`.